Fix boxcox to return same result for same input in one batch

yunwu52 opened the pull request:
Summary:
The SIMD path is using SLEEF version of pow which is slightly different from std::pow. The fix is to use the same vectorized code (with partial load and store) for the trailing data as well to ensure consistency between results.

Deploy: 
Need to make a hotfix in waas to monitor release signals, since this diff can cause testing failures in veloski and waas release correctness tests.

Test Plan: Sandcastle.

Differential Revision: D86218207

swolchok reviewed (approved):
Review automatically exported from Phabricator review in Meta.

yunwu52 committed:
[S564383] Reland D82265247 Fix boxcox to return same result for same input in one batch (#166986)

Summary:

Original commit changeset: 9d4b76544be0

Original Phabricator Diff: D82753931

Deploy: 
Need to make a hotfix in waas to monitor release signals, since this diff can cause testing failures in veloski and waas release correctness tests.

Test Plan: Sandcastle.

Reviewed By: swolchok

Differential Revision: D86218207
